src/catalogue/category/catalogueItemPropertiesMigrationDialog.component.test.tsx

```diff
@@ -0,0 +1,922 @@
+          unit: 'millimeters',
+          default_value: '2',
+          mandatory: false,
+        },
+        justModifyPropertyForm: false,
+      });
+
+      await user.click(screen.getByRole('button', { name: 'Finish' }));
+
+      expect(axiosPostSpy).toHaveBeenCalledWith(
+        '/v1/catalogue-categories/12/properties',
+        {
+          default_value: '2',
+          mandatory: false,
+          name: 'test',
+          type: 'string',
+          unit: 'millimeters',
+        }
+      );
+    });
+
+    it('adds a new property (type string and without default value)', async () => {
+      createView();
+      await modifyValues({
+        type: 'Add',
+        formField: {
+          name: 'test',
+          type: 'Text',
+          unit: 'millimeters',
+          mandatory: false,
+        },
+        justModifyPropertyForm: false,
+      });
+
+      await user.click(screen.getByRole('button', { name: 'Finish' }));
+
+      expect(axiosPostSpy).toHaveBeenCalledWith(
+        '/v1/catalogue-categories/12/properties',
+        {
+          mandatory: false,
+          name: 'test',
+          type: 'string',
+          unit: 'millimeters',
+        }
+      );
+    });
+
+    it('adds a new property (type boolean)', async () => {
+      createView();
+      await modifyValues({
+        type: 'Add',
+        formField: {
+          name: 'test',
+          type: 'Boolean',
+          default_value: 'false',
+          mandatory: true,
+        },
+        justModifyPropertyForm: false,
+      });
+
+      await user.click(screen.getByRole('button', { name: 'Finish' }));
+
+      expect(axiosPostSpy).toHaveBeenCalledWith(
+        '/v1/catalogue-categories/12/properties',
+        {
+          default_value: false,
+          mandatory: true,
+          name: 'test',
+          type: 'boolean',
+        }
+      );
+    });
+
+    it('uses labels to navigate through the stepper', async () => {
+      createView();
+      await user.click(
+        screen.getByLabelText(
+          'Select add to a new property or select edit to edit an existing property'
+        )
+      );
+
+      const dropdown = screen.getByRole('listbox', {
+        name: 'Select add to a new property or select edit to edit an existing property',
+      });
+      await user.click(
+        within(dropdown).getByRole('option', {
+          name: 'Add',
+        })
+      );
+
+      await user.click(screen.getByText('Add catalogue item property'));
+
+      expect(screen.getAllByLabelText('Property Name *').length).toEqual(1);
+      await user.click(screen.getByRole('button', { name: 'Back' }));
+      expect(screen.getAllByLabelText('Property Name *').length).toEqual(3);
+      await user.click(screen.getByText('Add catalogue item property'));
+
+      expect(screen.getAllByLabelText('Property Name *').length).toEqual(1);
+    });
+
+    it('displays errors when mandatory fields are not filled in and clears when text is updated', async () => {
```

Comment:
@joelvdavies he is a test were it checks for the for the default value when mandatory is true